### PR TITLE
Enable kafka cluster for installing hub-of-hubs

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: Meng <yanmxa@163.com>

### CODE REVIEW

**Comment by yanmxa** [comment]
https://github.com/stolostron/backlog/issues/23505

File: test/setup/kafka/kafka-operator.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  annotations:
+    olm.providedAPIs: Kafka.v1beta2.kafka.strimzi.io,KafkaBridge.v1beta2.kafka.strimzi.io,KafkaConnect.v1beta2.kafka.strimzi.io,KafkaConnectS2I.v1beta2.kafka.strimzi.io,KafkaConnector.v1beta2.kafka.strimzi.io,KafkaMirrorMaker.v1beta2.kafka.strimzi.io,KafkaMirrorMaker2.v1beta2.kafka.strimzi.io,KafkaRebalance.v1beta2.kafka.strimzi.io,KafkaTopic.v1beta2.kafka.strimzi.io,KafkaUser.v1beta2.kafka.strimzi.io
+  generateName: kafka-
```
Comment (morvencao) [inline test/setup/kafka/kafka-operator.yaml:6]: is the `generateName` must-have?

File: test/setup/kafka/kafka-operator.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  targetNamespaces:
+  - kafka
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: strimzi-kafka-operator
+  namespace: kafka
+spec:
+  channel: strimzi-0.23.x
```
Comment (clyang82) [inline test/setup/kafka/kafka-operator.yaml:19]: 0.23.x or 0.27.x?

File: test/setup/kafka/kafka_setup.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
+
```
Comment (clyang82) [inline test/setup/kafka/kafka_setup.sh:1]: You have missed `#!/bin/bash`

File: test/setup/kafka/kafka_setup.sh [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  operatorDeployed=$(kubectl -n kafka get Deployment/strimzi-cluster-operator-v0.23.0 --ignore-not-found)
+  while [ -z "$operatorDeployed" ]; do
+      echo "Waiting for strimzi-cluster-operator to become available"
+      sleep 10
+      operatorDeployed=$(kubectl -n kafka get Deployment/strimzi-cluster-operator-v0.23.0 --ignore-not-found)
+  done
+  kubectl -n kafka wait --for=condition=Available Deployment/strimzi-cluster-operator-v0.23.0 --timeout=600s
+
+  # deploy Kafka cluster CR
+  curl -s "https://raw.githubusercontent.com/stolostron/hub-of-hubs-kafka-transport/$branch/deploy/kafka-cluster.yaml" | kubectl apply -f -
```
Comment (clyang82) [inline test/setup/kafka/kafka_setup.sh:22]: Can we move `https://github.com/stolostron/hub-of-hubs-kafka-transport/blob/main/deploy/kafka-cluster.yaml` into this repo? `hub-of-hubs-kafka-transport` may be deprecated in the future.

File: test/setup/kafka/kafka_setup.sh [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+function deployKafka() {
+  # create namespace if not exists
+  kubectl create namespace kafka --dry-run=client -o yaml | kubectl apply -f -
+
+  # install community kafka operator
+  currentDir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+  kubectl apply -f ${currentDir}/kafka-operator.yaml
+    
+  # wait until operator is ready
+  operatorDeployed=$(kubectl -n kafka get Deployment/strimzi-cluster-operator-v0.23.0 --ignore-not-found)
```
Comment (clyang82) [inline test/setup/kafka/kafka_setup.sh:13]: can we define the version as an environment variable?

**Review by clyang82 — CHANGES_REQUESTED** [review]

File: test/setup/kafka/kafka-operator.yaml (hunk 1 above)
Comment (yanmxa) [reply test/setup/kafka/kafka-operator.yaml:6]: It's not necessary to add the generateName. I think we can remove it from the yaml file. It shouldn't make any difference.

File: test/setup/kafka/kafka_setup.sh (hunk 5 above)
Comment (yanmxa) [reply test/setup/kafka/kafka_setup.sh:13]: Yeah, I extracted the operator name with a specific version to the environment variable KAFKA_OPERATOR.

**Pull request merged**